### PR TITLE
Adding Windows 11 to the list of supported OS for self hosted runners.

### PR DESCRIPTION
**Pull request by MaxJPRey**
### Why:

Closes: 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Just adding Windows 11 in the list of supported OS for self hosted runners.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@MaxJPRey Thanks so much for opening a PR! I'll get this triaged for review ✨

File: content/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners.md [hunk 1]
```diff
@@ -109,6 +109,7 @@ The following operating systems are supported for the self-hosted runner applica
 - Windows 7 64-bit
 - Windows 8.1 64-bit
 - Windows 10 64-bit
+- Windows 11 64-bit
 - Windows Server 2012 R2 64-bit
```
Comment (SiaraMist) [inline content/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners.md:113]:
```suggestion
```

**Review by SiaraMist — DISMISSED** [review]
Thanks for adding this @MaxJPRey! While we're in here, I'm going to remove Windows Server 2012 R2 since it reached EOS in October, then I'll get this merged. 🚀

**Review by SiaraMist — APPROVED** [review]

**Comment by MaxJPRey** [comment]
@SiaraMist Thank you for removing Windows Server 2012 R2 and for merging the PR.

**Pull request merged**